File: MacOS/Type4Me/ASR/Providers/GoogleASRConfig.swift
import Foundation

struct GoogleASRConfig: ASRProviderConfig, Sendable {

    static let provider = ASRProvider.google
    static let displayName = "Google Cloud STT"

    static var credentialFields: [CredentialField] {[
        CredentialField(key: "serviceAccountJSON", label: "Service Account JSON", placeholder: L("粘贴 JSON 内容或文件路径", "Paste JSON content or file path"), isSecure: true, isOptional: false, defaultValue: ""),
    ]}

    let serviceAccountJSON: String

    init?(credentials: [String: String]) {
        guard let json = credentials["serviceAccountJSON"], !json.isEmpty else { return nil }
        self.serviceAccountJSON = json
    }

    func toCredentials() -> [String: String] {
        ["serviceAccountJSON": serviceAccountJSON]
    }

    var isValid: Bool { !serviceAccountJSON.isEmpty }
}

File: Windows/Type4Me/ASR/Providers/GoogleASRConfig.cs
using Type4Me.Localization;

namespace Type4Me.ASR.Providers;

public sealed class GoogleASRConfig
{
    public string ServiceAccountJSON { get; }

    public static ASRProvider Provider => ASRProvider.Google;
    public static string DisplayName => "Google Cloud STT";

    public static CredentialField[] CredentialFields =>
    [
        new() { Key = "serviceAccountJSON", Label = "Service Account JSON",
                Placeholder = Loc.L("粘贴 JSON 内容或文件路径", "Paste JSON content or file path"), IsSecure = true },
    ];

    private GoogleASRConfig(string json) { ServiceAccountJSON = json; }

    public static GoogleASRConfig? TryCreate(Dictionary<string, string> credentials)
    {
        if (!credentials.TryGetValue("serviceAccountJSON", out var json) || string.IsNullOrEmpty(json)) return null;
        return new GoogleASRConfig(json);
    }

    public Dictionary<string, string> ToCredentials() => new() { ["serviceAccountJSON"] = ServiceAccountJSON };
    public bool IsValid => !string.IsNullOrEmpty(ServiceAccountJSON);
}
